tests/5.0/teams_loop/test_target_teams_loop_allocate.c

```diff
@@ -0,0 +1,250 @@
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+/**
+  This is a basic test to demonstrate how allocate can be used with
+  target teams loop construct. Allocate clause is used to allocate a local
+  variable of size int.
+*/
+int testTargetTeamsAllocateCl() {
+  int a[N], b[N], c[N], local = 0;
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    c[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N])\
+        uses_allocators(omp_default_mem_alloc)\
+        allocate(omp_default_mem_alloc: local) private(local)
```

Comment:
> Please advise if we should remove the test file or any modification would allow allocate() clause still applicable.

A bit pointless, but you could add a 'firstprivate' clause (which then applies to target and teams, but not to 'loop') – and keep the 'private' for 'loop'.

But somewhat more sensible would be to split 'target teams loop' as follows – such that the private applies to 'teams' – That's still not very useful as the variable is twice privatized - once on 'teams' level and once on 'loop' level, but still.
```suggestion
#pragma omp target teams map(to: a[0:N], b[0:N]) map(from: c[0:N]) \
        uses_allocators(omp_cgroup_mem_alloc) \
        allocate(omp_low_lat_mem_alloc: local) private(local)
  #pragma omp loop private(local)
```

How about something like the following? Keeping the private but adding another firstprivatized variable, which is now put into "const" memory to make it more interesting than "default":
```C++
int magic_value = 42;
#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N]) \
        uses_allocators(omp_default_mem_alloc) \
        allocate(omp_const_mem_alloc: magic_value) firstprivate(magic_value) \
        private(local)
  for (int i = 0; i < N; i++) {
    local = a[i] + b[i] + magic_value;
    c[i] = local;
  }
```

Reply:
For me, the last option (use `first private(magic_value)`) seems to be the best.